src/lib/scheduler/node_queue_scheduler.cpp

```diff
@@ -188,7 +199,6 @@ void NodeQueueScheduler::_group_tasks(const std::vector<std::shared_ptr<Abstract
     if (!task->predecessors().empty() || !task->successors().empty()) {
       return;
```

Comment:
Side note: Should this be `continue` instead of `return`?

Reply:
I guess we assume here, that if for some task, the predecessors or successors are set, the given tasks already contain some kind of grouping and we can preemptively exit (to not mess anything up/create deadlocks). 

I am not 100% sure though.

Reply:
Not sure either, the comment appears to me as "skip this task" instead of "oh, seems like we are done with this set of tasks". It also appears that we could be stopping in the middle of the vector - if it is a "this vector is completely done or completely not-done" check, it would be clearer to check the first element before the loop and add a comment explaining why this is fair to do.

In the end this is not your change anyways, so no need to sweat it (although you did copy the pattern to your grouping :D)